netplan/configmanager.py

```diff
@@ -82,7 +82,7 @@ def parse(self, extra_config=None):
 
             self.np_state = libnetplan.State()
             self.np_state.import_parser_results(parser)
-        except libnetplan.LibNetplanException as e:
+        except Exception as e:
```

Comment:
suggestion (non-blocking): introduce a base class in libnetplan.py for our known exceptions and catch it here

The idea is again that exceptions not only signal user error (in the libnetplan.py exceptions) but also sometimes programmer error. For example, at some point some code could call this function like (which would seem reasonable given the available documentation):

```python
# inject eth42 in the existing configuration
manager.parse(extra_config={
    'network': {
        'ethernets': {
            'eth42': {}
        }
    })
```

I don't know what that would raise, but I'm betting it wouldn't be a libnetplan.* exception. IMHO, we want that exception to bubble up untouched to the caller.

(also, at some point we need to improve documentation here ;) )